### PR TITLE
Support for SGX simulation mode

### PR DESCRIPTION
**Pull request by bvavala**
This PR provides templates for ERCC attestation, from which Go source files can be generated for SGX hardware mode and simulation mode. In SIM mode, any required attestation check returns success.
Source generation is performed through makefiles and the C preprocessor.

This addresses issue #9 .

The sgx-enable patch addresses issue #63 and #64 : it passes tge sgx device to docker only if it exists; and, for DCAP, it passes the right device.

### CODE REVIEW

File: ercc/Makefile [hunk 1]
```diff
@@ -1,6 +1,28 @@
-.PHONY: all
+# Copyright 2019 Intel Corporation
```
Comment (mbrandenburger) [inline ercc/Makefile]: If you want to add this here, please also add IBM copyright as well as this file I have created already.

File: ercc/attestation/Makefile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+.PHONY: all clean
+
+IAS_GO_DEFINES=
+
+ifeq (${SGX_MODE}, HW)
```
Comment (mbrandenburger) [inline ercc/attestation/Makefile]: Can we treat SGX_MODE by default as HW?!

**Comment by mbrandenburger** [comment]
Thanks Bruno for this PR. I think this a good quick-and-dirty solution how we can enable simulation mode for the project. As we cannot simply use GOTAGS to compile ERCC in a way that it uses mocks to perform attestation verification; your approach seems to work. Alternately, one could have another ercc_sim chaincode that includes the mock verifier, and the user is responsible to install ercc_sim when needed.

File: ercc/attestation/Makefile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+IAS_GO_DEFINES+=-DUSE_SGX_HARDWARE_MODE
+endif
+
+GO_TARGETS=ias.go verifier.go
+
+all: check_sgx_mode $(GO_TARGETS)
+
+check_sgx_mode:
+ifndef SGX_MODE
```
[the remark this answers is not in the document]
Comment (bvavala) [reply ercc/attestation/Makefile]: I have added a default value in the top config.mk,
and also updated the sgx-enable patch with yours.

File: ercc/attestation/Makefile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	$(error SGX_MODE not defined)
+endif
+
+$(GO_TARGETS):
```
[the remark this answers is not in the document]
Comment (bvavala) [reply ercc/attestation/Makefile]: yes, it's better!
I have added a dependency
* on the templates
* on SGX_MODE

File: ercc/Makefile (hunk 1 above)
Comment (bvavala) [reply ercc/Makefile]: Marcus, I cannot for obvious reasons. But I am happy to wait for you to edit this file
* either directly in this PR and add a commit (you can do it throught the web interface, don't forget the signoff)
* or in another PR to be merged before or after this one.

File: ercc/attestation/Makefile (hunk 2 above)
Comment (bvavala) [reply ercc/attestation/Makefile]: Yes. I have added that in the top level config.mk

File: ercc/attestation/Makefile [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+ifeq (${SGX_MODE}, HW)
+IAS_GO_DEFINES+=-DUSE_SGX_HARDWARE_MODE
+endif
+
+TEMPLATE_SUFFIX=.template
+GO_TARGETS=ias.go verifier.go
+GO_TEMPLATES=$(addsuffix $(TEMPLATE_SUFFIX), $(GO_TARGETS))
+
+# put a dependency on the SGX_MODE environment variable
+ifneq "$(SGX_MODE)" "$(shell cat .sgx_mode 2>/dev/null)"
```
Comment (g2flyer) [inline ercc/attestation/Makefile]: Nice to capture also this dependency!

File: ercc/Makefile (hunk 1 above)
Comment (mbrandenburger) [reply ercc/Makefile]: ok, I created #72 which adds missing licenses.

File: README.md [hunk 6]
```diff
@@ -118,7 +118,8 @@ After installing the SGX SDK and SGX SSL double check that ``SGX_SDK`` and
 ``SGX_SSL`` variables are set correctly in your env. In particular,
 the makefiles are configured with meaningful defaults for variables
 like ``SGX_SDK``, ``SGX_ARCH``, ``SGX_MODE``, ``SGX_BUILD`` or
-``SGX_SSL``.  However, if you have non-standard values for install
+``SGX_SSL``. Notice: by default, ``SGX_MODE=HW``.
```
Comment (mbrandenburger) [inline README.md]: Can you also add `SGX_MODE=Sim` as an option here so users know what available options.

File: ercc/Makefile (hunk 1 above)
Comment (mbrandenburger) [reply ercc/Makefile]: so let's merge this first and then rebase.

File: ercc/attestation/Makefile [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+TOP = ../..
+include $(TOP)/build.mk
+
+.PHONY: all clean .sgx_mode
```
Comment (mbrandenburger) [inline ercc/attestation/Makefile]: .PHONY: all and clean are already in build.mk thus not needed here.

File: ercc/Makefile (hunk 1 above)
Comment (bvavala) [reply ercc/Makefile]: sounds good

File: README.md (hunk 6 above)
Comment (bvavala) [reply README.md]: I have rephrased it though... it was written immediately after...by you!! :)

File: ercc/attestation/Makefile (hunk 7 above)
Comment (bvavala) [reply ercc/attestation/Makefile]: Yep, it's redundant, I have removed it.

File: README.md (hunk 6 above)
Comment (mbrandenburger) [reply README.md]: by me ... 🙈

**Comment by mbrandenburger** [comment]
@bvavala seems that you have merge conflicts. I recommend to rebase your PR, squash all commits together, and then push force.

**Pull request merged**